docs/reference/query-builders.md

```diff
@@ -0,0 +1,382 @@
+
+`false` can be passed as an option to use *no sort definition at all*. This can be helpful when using MongoDB operators like `$near`, which already sort.
+
+If this method is never called or the argument is `undefined`, a case-insensitive sort on the title is normally the default. If `search()` has been called, then a sort by search result quality is the default.
+
+If the query sorts on a field that is defined in a doc type's schema with the `sortify: true` option, then this query builder will automatically substitute a "sortified" version of the field: case-insensitive and ignoring extra whitespace and punctuation. This provides a more natural sort than MongoDB normally does.
+
+### `skip()`
+
+```
+query.skip(10)
+```
+
+The `skip` builder accepts a number as an argument, then skips that number of documents in a query's results. This affects `toArray` and `toObject`. It does not affect `toDistinct` or `toMongo`.
+
+### `search()`
+
+```
+query.search('tree')
+```
+
+The `search` builder limits results to those matching the string passed as an argument. Search is implemented using MongoDB's `$text` operator and a full text index.
+
+If this query builder is set, the `sort` query builder will default to sorting by search quality. This is important because the worst of the full-text search matches will be of very poor quality.
+
+### `type()`
+
+```
+query.type('product')
+```
+
+The `type` builder can be used to limit a query to only one particular doc type. Pass the doc type name (string) as an argument.
+
+Generally you don't want to call this method directly. It will be better to include the doc type name as the `type` criteria in the original `find()` method arguments.
+
+### `withPublished()`
+
+```
+query.withPublished(true)
+```
+
+If the `withPublished` builder is set to `true`, then each document in the results will include a `_publishedDoc` property. That property will be set to the published version of that document. This is really only useful when querying for draft documents.
+
+## Page document query builders
+
+The following query builders only relate to page documents.
+
+### `ancestors()`
+
+```
+query.ancestors(true)
+
+query.ancestors({ children: true })
+```
+
+Setting the `ancestors` builder to `true` retrieves the array of "ancestors" for each returned page and assigns them to the `_ancestors` property. Page ancestors are those that precede a given page in the page tree hierarchy. The home page is `_ancestors[0]`. A returned page is not included in its own `_ancestors` array.
+
+If the builder argument is an object, do all of the above, and also call the query builders present in the object *on the query that fetches the ancestors*. For example, you can pass `{ children: true }` to fetch the children of each ancestor as the `_children` property of each ancestor, or pass `{ children: { depth: 2 } }` to get two layers of child pages.
```

Comment:
"If the builder's argument is an object, the builder does all of the above, and also calls..."